pkg/odo/cli/env/view.go

```diff
@@ -0,0 +1,107 @@
+package env
+
+import (
+	"fmt"
+	"os"
+	"reflect"
+	"text/tabwriter"
+
+	"github.com/openshift/odo/pkg/envinfo"
+	"github.com/openshift/odo/pkg/odo/genericclioptions"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	ktemplates "k8s.io/kubectl/pkg/util/templates"
+)
+
+const viewCommandName = "view"
+
+var (
+	viewLongDesc = ktemplates.LongDesc(`
+	View current values in odo environment file
+	`)
+
+	viewExample = ktemplates.Examples(`
+	# For viewing the current environment configuration settings
+	%[1]s
+	`)
+)
+
+// ViewOptions encapsulates the options for the command
+type ViewOptions struct {
+	context string
+	cfg     *envinfo.EnvSpecificInfo
+}
+
+// NewViewOptions creates a new ViewOptions instance
+func NewViewOptions() *ViewOptions {
+	return &ViewOptions{}
+}
+
+// Complete completes ViewOptions after they've been created
+func (o *ViewOptions) Complete(name string, cmd *cobra.Command, args []string) (err error) {
+	o.cfg, err = envinfo.NewEnvSpecificInfo(o.context)
+	if err != nil {
+		return errors.Wrap(err, "failed to load environment file")
+	}
+
+	return nil
+}
+
+// Validate validates the ViewOptions based on completed values
+func (o *ViewOptions) Validate() (err error) {
+	if !o.cfg.EnvInfoFileExists() {
+		return errors.Errorf("the context directory doesn't contain a component, please refer `odo create --help` to create a component")
+	}
+
+	return nil
+}
+
+// Run contains the logic for the command
+func (o *ViewOptions) Run() (err error) {
+	cs := o.cfg.GetComponentSettings()
+	w := tabwriter.NewWriter(os.Stdout, 5, 2, 2, ' ', tabwriter.TabIndent)
+	fmt.Fprintln(w, "PARAMETER NAME", "\t", "PARAMETER VALUE")
+	fmt.Fprintln(w, "Name", "\t", cs.Name)
+	fmt.Fprintln(w, "Namespace", "\t", cs.Namespace)
+	fmt.Fprintln(w, "DebugPort", "\t", showBlankIfNil(cs.DebugPort))
```

Comment:
Can we not print the fields that aren't set in the env? (which I guess could only be the namespace and debugport)

Reply:
I think that's a good indicator to let user know the current status of the fields that we support, for example from the view user would know the field is not set yet and odo do support the configuration of that field. This design applies to all the odo configuration commands like `odo config` and `odo preference`, so it would better we make all three odo configuration commands (`odo env`, `odo config`, `odo preference`) consistent.

Reply:
If that's that `odo preference` or `odo config` do, then I'm fine with that.